File: app/src/main/java/se/hactar/movieregister/model/MovieDao.java
package se.hactar.movieregister.model;


import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import java.util.List;

@Dao
public interface MovieDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAll(List<Movie> movies);

    @Query("SELECT * FROM movie")
    LiveData<List<Movie>> getAllLiveData();

    @Query("SELECT * FROM movie")
    List<Movie> getAll();

    @Query("SELECT * FROM movie WHERE imdb_id=:imdbId")
    Movie get(String imdbId);

    @Update
    void update(Movie movie);
}

File: app/src/main/java/se/hactar/movieregister/model/MovieDao.kt
package se.hactar.movieregister.model


import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import androidx.room.Update

@Dao
interface MovieDao {

    @get:Query("SELECT * FROM movie")
    val allLiveData: LiveData<List<Movie>>

    @get:Query("SELECT * FROM movie")
    val all: List<Movie>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAll(movies: List<Movie>)

    @Query("SELECT * FROM movie WHERE imdb_id=:imdbId")
    operator fun get(imdbId: String): Movie

    @Update
    fun update(movie: Movie)

    @Query("DELETE FROM movie")
    fun deleteAll()
}
